lib/puppet/provider/iis_application_pool/webadministration.rb

```diff
@@ -24,6 +24,7 @@ def create
     result   = self.class.run(inst_cmd)
     Puppet.err "Error creating apppool: #{result[:errormessage]}" unless result[:exitcode] == 0
     Puppet.err "Error creating apppool: #{result[:errormessage]}" unless result[:errormessage].nil?
+    @resource.provider = self.class.instances.find{|prov| prov.name == @resource[:name]}
```

Comment:
Considering that calls to Get-Website are extremely expensive what cost does this have running again without filtering by name in the powershell side?